### ISSUE TITLE
		module not found: eu.jrie.jetbrains#kotlin-shell-core;0.2.1

### ISSUE THREAD

**Issue by loxal** [jakubriegel/kotlin-shell#8]
```
root@if-alex-node-0:~# kotlin -version
Kotlin version 1.3.71-release-431 (JRE 14+36-1461)
root@if-alex-node-0:~# ./fsmoke-test.main.kts 
:: problems summary ::
:::: WARNINGS
		module not found: eu.jrie.jetbrains#kotlin-shell-core;0.2.1

	==== central: tried

	  https://repo1.maven.org/maven2/eu/jrie/jetbrains/kotlin-shell-core/0.2.1/kotlin-shell-core-0.2.1.pom

	  -- artifact eu.jrie.jetbrains#kotlin-shell-core;0.2.1!kotlin-shell-core.jar:

	  https://repo1.maven.org/maven2/eu/jrie/jetbrains/kotlin-shell-core/0.2.1/kotlin-shell-core-0.2.1.jar

error: cannot access script base class 'org.jetbrains.kotlin.mainKts.MainKtsScript'. Check your module classpath for missing or conflicting dependencies (fsmoke-test.main.kts:3:1)
error: unresolved reference: Repository (fsmoke-test.main.kts:3:7)
error: unresolved reference: DependsOn (fsmoke-test.main.kts:4:7)
error: unresolved reference: DependsOn (fsmoke-test.main.kts:5:7)
error: unresolved reference: CompilerOptions (fsmoke-test.main.kts:6:7)
error: unresolved reference: kotlinx (fsmoke-test.main.kts:7:13)
error: an annotation argument must be a compile-time constant (fsmoke-test.main.kts:7:13)
error: unresolved reference: shell (fsmoke-test.main.kts:29:1)
error: unresolved reference: stringLambda (fsmoke-test.main.kts:30:19)
error: unresolved reference: it (fsmoke-test.main.kts:30:34)
error: unresolved reference: pipeline (fsmoke-test.main.kts:31:5)
error: unresolved reference: process (fsmoke-test.main.kts:31:62)
error: unresolved reference: process (fsmoke-test.main.kts:31:93)
fsmoke-test.main.kts: error: unresolved dependency: eu.jrie.jetbrains#kotlin-shell-core;0.2.1: not found
fsmoke-test.main.kts:3:1: error: cannot access script base class 'org.jetbrains.kotlin.mainKts.MainKtsScript'. Check your module classpath for missing or conflicting dependencies
@file:Repository("https://dl.bintray.com/jakubriegel/kotlin-shell")
^
fsmoke-test.main.kts:3:7: error: unresolved reference: Repository
@file:Repository("https://dl.bintray.com/jakubriegel/kotlin-shell")
      ^
fsmoke-test.main.kts:4:7: error: unresolved reference: DependsOn
@file:DependsOn("eu.jrie.jetbrains:kotlin-shell-core:0.2.1")

```
 
...comes from this code...

```
#!/usr/bin/env kotlin

@file:Repository("https://dl.bintray.com/jakubriegel/kotlin-shell")
@file:DependsOn("eu.jrie.jetbrains:kotlin-shell-core:0.2.1")

import eu.jrie.jetbrains.kotlinshell.shell.*

shell {
    val toUpper = stringLambda { it.toUpperCase() to "" }
    pipeline { "curl https://uli-ifinder-demo.intrafind.net".process() pipe "grep Analyzer".process() pipe toUpper }
}
System.exit(0)
```
...whereas...

```
@file:Repository("https://jcenter.bintray.com")
@file:DependsOn("org.slf4j:slf4j-simple:1.7.30")
@file:DependsOn("com.squareup.okhttp3:okhttp:4.4.1")
@file:CompilerOptions("-Xopt-in=kotlin.RequiresOptIn")

```
...works fine.

**Comment by jakubriegel** [comment]
Hi! I seems to be a problem with `kotlin-main-kts` dependency resolver. For a quick solution a would recommend using local jar of `kotlin-shell`. So the scrript would look like:
```kotlin
//@file:Repository("https://dl.bintray.com/jakubriegel/kotlin-shell")
//@file:DependsOn("eu.jrie.jetbrains:kotlin-shell-core:0.2.1")

@file:DependsOn("kotlin-shell/kotlin-shell-core/build/libs/kotlin-shell-core-0.2.1.jar")
@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.3")
@file:DependsOn("org.jetbrains.kotlinx:kotlinx-io-jvm:0.1.11")
@file:DependsOn("org.zeroturnaround:zt-exec:1.11")
@file:DependsOn("org.slf4j:slf4j-nop:1.7.28")
@file:CompilerOptions("-Xopt-in=kotlin.RequiresOptIn")
@file:OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)


import eu.jrie.jetbrains.kotlinshell.shell.*
import kotlin.system.exitProcess

shell {
    val toUpper = stringLambda { it.toUpperCase() to "" }
    pipeline { "curl https://uli-ifinder-demo.intrafind.net".process() pipe "grep Analyzer".process() pipe toUpper }
}
exitProcess(0)
```

To get the jar you can clone `kotlin-shell` repo and run `./gradlew shadowJar` or download it from [bintray](https://bintray.com/beta/#/jakubriegel/kotlin-shell/kotlin-shell-core?tab=files).

Meanwhile I will try to investigate this problem further

**Comment by loxal** [comment]
Thanks a lot, your proposal solved my problem.

The curious fact is that the script runs fine with
```
@file:Repository("https://dl.bintray.com/jakubriegel/kotlin-shell")
@file:DependsOn("eu.jrie.jetbrains:kotlin-shell-core:0.2.1")
```

on my day-to-day notebook with Ubuntu 20.04 and a regular system user that has a quite full local `~/.gradle` repository.

However on my CI where the job is executed as root, I had to deploy your fix which worked fine. I hope, this info helps you fixing the issue.

**Comment by loxal** [comment]
test.main.kts:

```
#!/usr/bin/env kotlin

@file:Repository("https://jcenter.bintray.com")
//@file:Repository("https://dl.bintray.com/kotlin/kotlinx")
//@file:Repository("https://dl.bintray.com/jakubriegel/kotlin-shell")
//@file:Repository("central", "central", "https://dl.bintray.com")
@file:DependsOn("eu.jrie.jetbrains:kotlin-shell-core:0.2.1")
//@file:DependsOn("eu.jrie.jetbrains:kotlin-shell-core:0.2-dev")
//@file:DependsOn("kotlin-shell-core-0.2.1.jar")

@file:DependsOn("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.5")
@file:DependsOn("io.ktor:ktor-http-jvm:1.3.1")
@file:DependsOn("org.jetbrains.kotlin:kotlin-test-junit5:1.3.72")
@file:DependsOn("org.zeroturnaround:zt-exec:1.11")
@file:DependsOn("org.slf4j:slf4j-nop:1.7.30")

@file:DependsOn("com.squareup.okhttp3:okhttp:4.5.0")
@file:CompilerOptions("-Xopt-in=kotlin.RequiresOptIn")

import eu.jrie.jetbrains.kotlinshell.shell.shell
import io.ktor.http.HttpStatusCode
import okhttp3.OkHttpClient
import okhttp3.Request
import kotlin.system.exitProcess
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

println("TENANT ${System.getenv("TENANT")}")

lateinit var targetUrl: String
println("args.size: ${args.size}")
if (args.isNotEmpty()) {
    println("args: ${args.joinToString()}")
    targetUrl = args[0]
} else {
    println("targetUrl required")
    exitProcess(2)
}
println("targetUrl $targetUrl")

@kotlinx.coroutines.ExperimentalCoroutinesApi
val httpClient = OkHttpClient.Builder()
        .followRedirects(false)
        .followSslRedirects(false)
        .build()

val request = Request.Builder()
        .url(targetUrl)
        .build()
@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)
httpClient.newCall(request).execute().use { response ->
    println(response.body?.contentLength())
    println(response.code)
    assertNotNull(response.code)
    assertEquals(200, response.code)
    assertEquals(HttpStatusCode.OK.value, response.code)
}

@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)
shell {
    val toUpper = stringLambda { it.toUpperCase() to "" }
    pipeline { "curl $targetUrl".process() pipe "grep Analyzer".process() pipe toUpper }
}

exitProcess(0)

```
Run this file locally

`./test.main.kts https://example.com`

...everything runs fine with Java 13.

To reproduce the issue, run

 `docker run -it -v /home/alex/Desktop:/srv loxal/kotlin-toolbox:1.4-M1-java13 ./test.main.kts`

Kotlin v1.3.x prints the same error message as Kotlin v1.4-M1. Apparently the container environment creates the virgin environments that causes the problem. 

cc @jakubriegel

**Comment by jakubriegel** [comment]
Should now be fixed by #16
